doc/source/redis-memory-management.rst

```diff
@@ -0,0 +1,103 @@
+Redis Memory Management (EXPERIMENTAL)
+==============================================
+
+Ray stores metadata associated to tasks and objects in one or more Redis
+servers, as described in :doc:`internals-overview`.  Applications that are
+long-running or have high task/object generation rate could risk high memory
+pressure, potentially leading to out-of-memory (OOM) errors.
+
+Here, we describe an experimental feature that transparently flushes metadata
+entries out of Redis memory.
+
+Requirements
+-------------
+
+As of early July 2018, the automatic memory management feature requires building
+Ray from source.  We are planning on eliminating this step in the near future by
+releasing official wheels.
+
+Building Ray
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, follow `instructions to build Ray from source
+<installation.html#building-ray-from-source>`__ to install prerequisites.  After
+the prerequisites are installed, instead of doing the regular ``pip install`` as
+referenced in that document,
+
+.. code-block:: bash
+
+  git clone https://github.com/ray-project/ray.git
+  cd ray/python
+  pip install -e . --verbose  # Add --user if you see a permission denied error.
+
+Pass these special flags instead:
+
+.. code-block:: bash
+
+  env RAY_USE_NEW_GCS=on RAY_USE_XRAY=1 pip install -e . --verbose
+
+Running Ray applications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+At run time the environment variables ``RAY_USE_NEW_GCS=on`` and
+``RAY_USE_XRAY=1`` are also required.
+
+.. code-block:: bash
+
+  # Export once instead of using "env" on every command.
+  export RAY_USE_NEW_GCS=on
+  export RAY_USE_XRAY=1
+  python my_ray_script.py  # Or launch python/ipython.
+
+Flush Policy
+-----------------
+
+After building Ray using the method above, simply add these two lines after
+``ray.init()`` to activate automatic memory flushing:
+
+.. code-block:: python
+
+   ray.init(...)
+
+   policy = ray.experimental.SimpleGcsFlushPolicy()
+   ray.experimental.set_flushing_policy(policy)
+
+   # My awesome Ray application logic follows.
+
+Paramaters of the flushing policy
+--------------------------------------
+
+There are three `user-configurable parameters
+<https://github.com/ray-project/ray/blob/master/python/ray/experimental/gcs_flush_policy.py#L31>`_
```

Comment:
Use a permanent link because this will break if we modify that file.

Reply:
Done